includes/Modules/Analytics_4/Conversion_Reporting/Conversion_Reporting_Events_Sync.php

```diff
@@ -73,17 +97,44 @@ public function sync_detected_events() {
 			return;
 		}
 
+		// Get current stored detected events.
+		$settings              = $this->settings->get();
+		$saved_detected_events = isset( $settings['detectedEvents'] ) ? $settings['detectedEvents'] : array();
+
 		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 		if ( empty( $report->rowCount ) ) {
 			$this->settings->merge( array( 'detectedEvents' => array() ) );
 
+			if ( ! empty( $this->transients->get( static::DETECTED_EVENTS_TRANSIENT ) ) ) {
+				$this->transients->delete( static::DETECTED_EVENTS_TRANSIENT );
+			}
+
+			if ( ! empty( $saved_detected_events ) ) {
+				$this->transients->set( static::LOST_EVENTS_TRANSIENT, $saved_detected_events );
+			}
+
 			return;
 		}
 
 		foreach ( $report->rows as $row ) {
 			$detected_events[] = $row['dimensionValues'][0]['value'];
 		}
 
+		$new_events  = array_diff( $detected_events, $saved_detected_events );
+		$lost_events = array_diff( $saved_detected_events, $detected_events );
+
+		if ( ! empty( $new_events ) ) {
+			$this->transients->set( 'googlesitekit_conversion_reporting_detected_events', array_values( $new_events ) );
+		}
+
+		if ( ! empty( $lost_events ) ) {
+			$this->transients->set( 'googlesitekit_conversion_reporting_lost_events', array_values( $lost_events ) );
+		}
+
+		if ( empty( $saved_detected_events ) ) {
+			$this->transients->set( 'googlesitekit_conversion_reporting_detected_events', $detected_events );
+		}
```

Comment:
Use constants instead of hardcoded transient names.

Reply:
@eugene-manuilov this has been addressed in the latest 2 or so commits.